platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,311 @@
+   */
+  @NonNull
+  String getStateKey();
+
+  /**
+   * Gets the singleton value.
+   *
+   * @return The value, or null if there is no value.
+   */
+  @Nullable
+  T get();
+
+  /**
+   * Gets whether the value of this {@link ReadableSingletonState} has been read.
+   *
+   * @return true if {@link #get()} has been called on this instance
+   */
+  boolean isRead();
+}
+/**
+ * Provides mutable access to singleton state.
+ *
+ * @param <T> The type of the state
+ */
+public interface WritableSingletonState<T> extends ReadableSingletonState<T> {
+  /**
+   * Sets the given value on this state.
+   *
+   * @param value The value. May be null.
+   */
+  void put(@Nullable T value);
+
+  /**
+   * Gets whether the {@link #put(Object)} method has been called on this instance.
+   *
+   * @return True if the {@link #put(Object)} method has been called
+   */
+  boolean isModified();
+}
+```
+
+## Requirements
+
+- `PlatformState` should be accessible via the State API 
+- there should be a new module - `swirlds-state-api` - to host a set of interfaces, records, and abstract classes that represent the Hashgraph state.
+  This module should have a minimal set of dependencies. The Block Node should not have a compile-time dependency
+  on any other modules but this one to interact with the state.
+
+## Changes
+
+### Public API
+
+State API classes and interfaces will migrate to a designated module. The classes themselves will not change. `PlatformState` class will become a part of the public API.
+
+Usage example:
+```java
+private PlatformState findPlatformState(HederaState state) {
+  final ReadableStates states = state.getReadableStates(PlatformState.NAME);   
+  final ReadableSingletonState<PlatformState> platformState = states.getSingleton(PlatformState.PLATFORM_STATE_KEY);
+  return platformState.get();
+}
+```
+
+## Components and Architecture
+
+[### Platform State](#platform-state)
+
+Currently `PlatformState` is a special case node of the merkle tree. That is, it doesn't conform to State API. 
+As a part of the simplification effort **it should be refactored to a singleton state**. 
+As a part of this change, `PlatformState` should not be a `MerkleNode` anymore.
+
+Protobuf definition:
+
+```
+/**
+ * The current state of platform consensus.<br/>
+ * This message stores the current consensus data for the platform
+ * in network state.
+ *
+ * The platform state SHALL represent the latest round's consensus.<br/>
+ * This data SHALL be used to ensure consistency and provide critical data for
+ * restart and reconnect.
+ */
+message PlatformState {
+    /**
+     * A consensus round.<br/>
+     * The round represented by this state.
+     * <p>
+     * This message SHALL represent the network state after handling of all
+     * transactions reaching consensus in all previous rounds.<br/>
+     * The first state (genesis state) SHALL have a round of 0.<br/>
+     * The first round to handle transactions SHALL be round `1`.
```

Comment:
I think this comment has an off-by-one error. A state with round 10 means all transactions in rounds 0 through 10 have been applied. The statement `This message SHALL represent the network state after handling of all
     transactions reaching consensus in all previous rounds` misses the current round.

Reply:
Fair enough. Lazar recommended removing this field in favor of the one that is in `ConsensusSnapshot`, so this comment is no longer there.

Reply:
Yeah, when we handle round 7, we've handled all transactions up to and including the transactions within round 7.